rocAL/include/api/rocal_api_types.h

```diff
@@ -469,6 +469,33 @@ struct RocalShardingInfo {
           stick_to_shard(true),
           shard_size(-1)
     {}
+
+    // Constructor that initializes all members
+    RocalShardingInfo(
```

Comment:
Do we need to introduce this constructor? Default constructor will assign values which are passed and only in case where no values are passed it will assign the default values ryt?

Reply:
This is used in the rocal_api_data_loaders.cpp file to initialize the members.